### PR TITLE
gui-wm/hyprland: add 0.30.0

### PR DESCRIPTION
**Pull request by MrRoy**

### CODE REVIEW

**Review by arthurzam — COMMENTED** [review]
Fails to build for me:

```
FAILED: hyprctl/hyprctl.p/main.cpp.o 
x86_64-pc-linux-gnu-g++ -Ihyprctl/hyprctl.p -Ihyprctl -I../hyprland-source/hyprctl -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wno-unused-parameter -Wno-unused-value -Wno-missing-field-initializers -Wno-narrowing '-DGIT_BRANCH=""' '-DGIT_COMMIT_HASH=""' '-DGIT_COMMIT_MESSAGE=""' '-DGIT_DIRTY="dirty"' -DNO_XWAYLAND -DUSES_SYSTEMD -std=c++23 -O2 -pipe -frecord-gcc-switches -MD -MQ hyprctl/hyprctl.p/main.cpp.o -MF hyprctl/hyprctl.p/main.cpp.o.d -o hyprctl/hyprctl.p/main.cpp.o -c ../hyprland-source/hyprctl/main.cpp
../hyprland-source/hyprctl/main.cpp:15:10: fatal error: format: No such file or directory
   15 | #include <format>
      |          ^~~~~~~~
compilation terminated.
```

`gcc (Gentoo 12.2.1_p20221008 p1) 12.2.1 20221008`

`[ebuild  N     ] gui-wm/hyprland-0.30.0::gentoo  USE="systemd -X -legacy-renderer" VIDEO_CARDS="-nvidia" 0 KiB`

**Comment by thesamesam** [comment]
I think std::format needs GCC 13. But you also really shouldn't be using such an old version of GCC 12. There's many serious bugs fixed since, like miscompilations.

**Comment by arthurzam** [comment]
> I think std::format needs GCC 13. But you also really shouldn't be using such an old version of GCC 12. There's many serious bugs fixed since, like miscompilations.

Yeah, I agree my gcc version is old, which is why I'm currently building 13 on my system. But since we still support 12, shouldn't we add check for minimal gcc version (or minimal clang version), since we use here c++23.

**Comment by thesamesam** [comment]
Yeah, we should. But to be clear, I just mean it's fine to use GCC 12, just not an ancient version _of_ GCC 12. If you're struggling with GCC build times, please: 1) try https://github.com/gentoo/gentoo/pull/33114; 2) maybe use the new binhost's binpkgs for gcc, and/or 3) build it with EXTRA_ECONF="--disable-bootstrap" (not ideal but better than ancient gcc).

File: gui-wm/hyprland/hyprland-0.30.0.ebuild [hunk 1]
```diff
@@ -77,6 +77,11 @@ BDEPEND="
 	virtual/pkgconfig
 "
 
+PATCHES=(
+	"${FILESDIR}/hyprland-0.30.0-no-wlroots-automagic-r1.patch"
+	"${FILESDIR}/hyprland-0.30.0-3400-fix-build.patch"
+)
+
 pkg_setup() {
```
Comment (thesamesam) [inline gui-wm/hyprland/hyprland-0.30.0.ebuild:85]: Please update the check in pkg_setup to need GCC 13 because of std::format.

Comment (MrRoy) [reply gui-wm/hyprland/hyprland-0.30.0.ebuild:85]: Done, thanks !

**Review by arthurzam — APPROVED** [review]

**Pull request closed without merging**